cmd/lotus-shed/indexes.go

```diff
@@ -879,3 +900,339 @@ var backfillTxHashCmd = &cli.Command{
+		srv, err := lcli.GetFullNodeServices(cctx)
+		if err != nil {
+			return err
+		}
+		defer srv.Close() //nolint:errcheck
+
+		api := srv.FullNodeAPI()
+		ctx := lcli.ReqContext(cctx)
+
+		basePath, startHeight, err := parsePruneArgs(ctx, cctx, api)
+		if err != nil {
+			return xerrors.Errorf("error parsing prune args: %w", err)
+		}
+
+		if err := pruneTxIndex(ctx, api, basePath, startHeight); err != nil {
+			return xerrors.Errorf("error pruning txindex: %w", err)
+		}
+
+		return nil
+	},
+}
+
+var pruneEventsCmd = &cli.Command{
+	Name:  "prune-events",
+	Usage: "Prune the events.db for events older than a given height",
+	Flags: []cli.Flag{
+		&cli.IntFlag{
+			Name:  "older-than",
+			Value: 0,
+			Usage: "height to start pruning events older than this epoch, 0 means start from head",
+		},
+	},
+	Action: func(cctx *cli.Context) error {
+		srv, err := lcli.GetFullNodeServices(cctx)
+		if err != nil {
+			return err
+		}
+		defer srv.Close() //nolint:errcheck
+
+		api := srv.FullNodeAPI()
+		ctx := lcli.ReqContext(cctx)
+
+		basePath, startHeight, err := parsePruneArgs(ctx, cctx, api)
+		if err != nil {
+			return xerrors.Errorf("error parsing prune args: %w", err)
+		}
+
+		if err := pruneEventsIndex(ctx, basePath, startHeight); err != nil {
+			return xerrors.Errorf("error pruning events index: %w", err)
+		}
+
+		return nil
+	},
+}
+
+// pruneAllIndexesCmd is a command that prunes all the indexes
+var pruneAllIndexesCmd = &cli.Command{
+	Name:  "prune-all-indexes",
+	Usage: "Prune all the indexes DBs (msgindex, txhash, events) older than a given height",
+	Flags: []cli.Flag{
+		&cli.IntFlag{
+			Name:  "older-than",
+			Value: 0,
+			Usage: "height to start pruning events older than this epoch, 0 means start from head",
+		},
+	},
+	Action: func(cctx *cli.Context) error {
+		srv, err := lcli.GetFullNodeServices(cctx)
+		if err != nil {
+			return err
+		}
+		defer srv.Close() //nolint:errcheck
+
+		api := srv.FullNodeAPI()
+		ctx := lcli.ReqContext(cctx)
+
+		basePath, startHeight, err := parsePruneArgs(ctx, cctx, api)
+		if err != nil {
+			return xerrors.Errorf("error parsing prune args: %w", err)
+		}
+
+		var g = new(errgroup.Group)
+
+		g.Go(func() error {
+			if err := pruneMsgIndex(ctx, basePath, startHeight); err != nil {
+				return xerrors.Errorf("error pruning msgindex: %w", err)
+			}
+
+			return nil
+		})
+
+		g.Go(func() error {
+			if err := pruneTxIndex(ctx, api, basePath, startHeight); err != nil {
+				return xerrors.Errorf("error pruning txindex: %w", err)
+			}
+
+			return nil
+		})
+
+		g.Go(func() error {
+			if err := pruneEventsIndex(ctx, basePath, startHeight); err != nil {
+				return xerrors.Errorf("error pruning events index: %w", err)
+			}
+
+			return nil
+		})
+
+		return g.Wait()
+	},
+}
+
+// pruneEventsIndex is a helper function that prunes the events index.db for a number of epochs starting from a specified height
+func pruneEventsIndex(_ context.Context, basePath string, startHeight int64) error {
```

Comment:
just ditch the context arg if you have no way of using it, I know it's nice to have symmetry but if you don't need the symmetry for any tactical reason then just keep it clean

Reply:
Done.